### ISSUE TITLE
What does this library do?

### ISSUE THREAD

**Issue by jonathanstrong** [mcgoo/openonload-rs#1]
Hey, 

I'm in the process of trying to figure out onload in general, and my code base is in rust, and I happened across this. What purpose does it serve to have rust bindings to onload? Any information you could provide would be appreciated. Thank you!

**Comment by mcgoo** [comment]
It's an automatically generated wrapper around the [etherfabric api](https://github.com/majek/openonload/blob/master/README.ef_vi), which is a non socket based zero-copy api for pulling packets off the wire that offers even better performance. Although I have used the etherfabric stuff from C/C++, I never got around to using it from Rust so I can't guarantee that this lib works but it builds. (The API is straight C which bindgen tends to do a good job with, so it's probably going to work.)  

There is another header `onload/extensions.h` which I did not include (for no particular reason), that controls onload itself. It allows an application to determine whether it is running under onload, and select on a per socket basis whether to perform acceleration or not. That's useful if you are running out of hardware filters, or if you have a lot of less latency sensitive connections in addition to the sockets that you want accelerated.

There is more documentation in ef_vi.pdf on support.solarflare.com under solarflare->linux->openonload. Requires registration - I couldn't work out how to link it.
